### PR TITLE
Rename topics to tags.

### PR DESCRIPTION
**Pull request by wwajerowicz**

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/output/annotation/DescriptionAnnotation.java [hunk 1]
```diff
@@ -18,8 +18,11 @@
     @Override
     public void write(T entity, FieldWriter writer, OutputContext ctxt) throws IOException {
         writer.writeObject(publisherWriter, entity.getSource(), ctxt);
-        if (entity instanceof Topic) {
-            writer.writeField("topic_type", ((Topic) entity).getType()); 
+        if (entity instanceof Tag) {
+            Tag tag = (Tag) entity;
+            writer.writeField("namespace", tag.getNamespace());
+            writer.writeField("value", tag.getValue());
+            writer.writeField("topic_type", tag.getType());
```
Comment (mbst-jamie) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/DescriptionAnnotation.java]: tag_type?

File: atlas-api/src/main/java/org/atlasapi/output/annotation/TagsAnnotation.java [hunk 2]
```diff
@@ -21,50 +21,50 @@
...
-    private static final Function<TopicRef, Id> REF_TO_ID = new Function<TopicRef, Id>() {
+    private static final Function<TagRef, Id> REF_TO_ID = new Function<TagRef, Id>() {
         @Override
-        public Id apply(TopicRef input) {
+        public Id apply(TagRef input) {
             return input.getTopic();
         }
     };
-    private static Function<Topic, Id> TOPIC_ID = new Function<Topic, Id>() {
+    private static Function<Tag, Id> TOPIC_ID = new Function<Tag, Id>() {
```
Comment (mbst-jamie) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/TagsAnnotation.java]: TAG_ID?

File: atlas-api/src/main/java/org/atlasapi/output/annotation/TagsAnnotation.java [hunk 3]
```diff
@@ -21,50 +21,50 @@
...
-    private final EntityWriter<Topic> topicWriter;
+    private final TagResolver tagResolver;
+    private final EntityWriter<Tag> topicWriter;
     
-    public TopicsAnnotation(TopicResolver topicResolver, EntityWriter<Topic> topicListWriter) {
+    public TagsAnnotation(TagResolver tagResolver, EntityWriter<Tag> topicListWriter) {
         super();
-        this.topicResolver = topicResolver;
+        this.tagResolver = tagResolver;
         this.topicWriter = topicListWriter;
```
Comment (mbst-jamie) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/TagsAnnotation.java]: tagListWriter

**Comment by mbst-jamie** [comment]
So we also need to rename the vars/methods/fields etc, guessing you only renamed the type? Do a find and replace of the word topic for tag across the repo and see what it breaks/see if the tests still pass.

**Comment by tfm** [comment]
:+1:

**Pull request merged**